ui/src/app/pages/data/data-set/export-data-set-modal.tsx

```diff
@@ -119,19 +119,28 @@ class ExportDataSetModal extends React.Component<
       cohortIds: dataSet.cohorts.map(c => c.id),
       domainValuePairs: dataSet.domainValuePairs
     };
-    await dataSetApi().exportToNotebook(
-      workspaceNamespace, workspaceFirecloudName,
-      {
-        dataSetRequest: request,
-        notebookName: this.state.notebookName,
-        newNotebook: this.state.newNotebook,
-        kernelType: this.state.kernelType
-      });
-    // Open notebook in a new tab and close the modal
-    const notebookUrl = `/workspaces/${workspaceNamespace}/${workspaceFirecloudName}/notebooks/
-      preview/${appendNotebookFileSuffix(encodeURIComponentStrict(this.state.notebookName))}`;
-    navigateByUrl(notebookUrl);
-    this.props.closeFunction();
+    try {
+      await dataSetApi().exportToNotebook(
+        workspaceNamespace, workspaceFirecloudName,
+        {
+          dataSetRequest: request,
+          notebookName: this.state.notebookName,
+          newNotebook: this.state.newNotebook,
+          kernelType: this.state.kernelType
+        });
+      // Open notebook in a new tab and close the modal
+      const notebookUrl = `/workspaces/${workspaceNamespace}/${workspaceFirecloudName}` +
+        `/notebooks/preview/${appendNotebookFileSuffix(
+          encodeURIComponentStrict(this.state.notebookName))}`;
+      navigateByUrl(notebookUrl);
+      this.props.closeFunction();
+    } catch (error) {
+      // https://precisionmedicineinitiative.atlassian.net/browse/RW-3782
+      // The exportToNotebook call can fail with a 400.  Catch that error to ensure the user can exit the modal.
+      // TODO: better failure UX
```

Comment:
Some quick ideas at EOW, for further discussion:

We don't have a real workaround for the user, but we could give them more information with some error text.

We could also close the modal.  But maybe there is a possibility of success later?  Keeping it open lets the user try again.

Reply:
Keeping the modal open and displaying an error message somewhere on screen (e.g. at the top) is probably the ideal flow here. We'd want to develop standard components / UI patterns to support that though.

Reply:
Agreed, as a longer term improvement outside this PR.